File: Difficulty/A. Easy/0344-reverse-string/0344-reverse-string-TwoPointerApproach.py
class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        """
        Reverses the input list `s` in-place. This function does not return anything.
        
        Parameters:
        s (List[str]): A list of characters to be reversed in-place.
        
        Returns:
        None: The function modifies the input list in-place and does not return anything.
        """
        # Initialize two pointers, one starting from the beginning (left)
        # and the other starting from the end (right) of the list
        left, right = 0, len(s) - 1

        # Swap elements at the left and right pointers until they meet in the middle
        while left < right:
            # Swap the elements at the two pointers
            s[left], s[right] = s[right], s[left]
            
            # Move the pointers towards the middle
            left += 1
            right -= 1

File: Explore Card Track/5. Recursion I/0344-reverse-string/0344-reverse-string-TwoPointerApproach.c
/**
 * @brief Reverses the input string in-place using the two-pointer technique.
 *
 * @param s (char*): Pointer to the string (character array).
 * @param sSize (int): The size of the string.
 */
void reverseString(char* s, int sSize) {
    // Initialize two pointers: one at the start and one at the end of the string
    int left = 0;
    int right = sSize - 1;

    // Loop until the left pointer is greater than or equal to the right pointer
    while (left < right) {
        // Swap the characters at the left and right indices
        char temp = s[left];
        s[left] = s[right];
        s[right] = temp;

        // Move the left pointer towards the center
        left++;
        // Move the right pointer towards the center
        right--;
    }
}
